internal/backend/sessions/sessionworkflows/workflow.go

```diff
@@ -375,25 +372,30 @@ func (w *sessionWorkflow) waitOnFirstSignal(wctx workflow.Context, signals []str
 	return signalID, nil
 }
 
-func (w *sessionWorkflow) getNextEvent(ctx context.Context, signalID string) (map[string]sdktypes.Value, error) {
+func (w *sessionWorkflow) getNextEvent(ctx context.Context, signalID uuid.UUID) (map[string]sdktypes.Value, error) {
 	wctx := sessioncontext.GetWorkflowContext(ctx)
 
 	var signal scheme.Signal
 	if err := workflow.ExecuteLocalActivity(wctx, w.ws.svcs.DB.GetSignal, signalID).Get(wctx, &signal); err != nil {
 		w.z.Panic("get signal", zap.Error(err))
 	}
 
-	iid := sdktypes.NewIDFromUUID[sdktypes.IntegrationID](signal.Connection.IntegrationID)
-	cid := sdktypes.NewIDFromUUID[sdktypes.ConnectionID](&signal.Connection.ConnectionID)
+	var did sdktypes.EventDestinationID
+	if signal.ConnectionID != nil {
+		did = sdktypes.NewEventDestinationID(sdktypes.NewIDFromUUID[sdktypes.ConnectionID](signal.ConnectionID))
+	} else if signal.TriggerID != nil {
+		did = sdktypes.NewEventDestinationID(sdktypes.NewIDFromUUID[sdktypes.TriggerID](signal.TriggerID))
+	} else {
+		return nil, fmt.Errorf("invalid signal %v: no connection or trigger ids, got %v", signalID, signal.DestinationID)
+	}
```

Comment:
can't we just list by connection_id or trigger_id and get the first ? instead of using a destination_id ?

Reply:
that'll will make twice the calls. this is just translating from a stored data to the actual type.